package/gluon.mk

```diff
@@ -20,12 +20,31 @@ shell-unescape = $(subst @1,@,$(subst @2,$(newline),$(1)))
 shell-verbatim = $(call shell-unescape,$(call shell-escape,$(1)))
 
 
+ifeq (${GLUON_TEST_PKG},1)
+
+define GluonCheckSite
+printf "%s" "Testing check_site.lua: "
+export IPKG_INSTROOT=/
+cat > /tmp/check_site_lib.lua <<'END__GLUON__CHECK__SITE__LIB'
+$(call shell-verbatim,cat '${TOPDIR}/../scripts/check_site.lua')
+END__GLUON__CHECK__SITE__LIB
+lua /tmp/check_site_lib.lua <<'END__GLUON__CHECK__SITE'
+$(call shell-verbatim,cat '$(1)')
+END__GLUON__CHECK__SITE
+rm /tmp/check_site_lib.lua
+echo ok
+endef
+
+else
+
```

Comment:
Hmm, I'm don't like this change... I'll see if I can come up with an alternative solution.

Reply:
Please let me know when you have decided what to do here.

Or maybe you could clarify why you don't like the approach? Then I could think about a solution too.

Reply:
It has been a month. Is this something that we can address another time or does it prevent merging this?

Reply:
And two more weeks went by. :disappointed:

Reply:
@neoraider As this becomes a rather stuck feedback-loop; how about meeting up in mumble for a few minutes tomorrow evening? Maybe that makes it easier to brainstorm and clearer for @lemoer what benefits the alternative solution should provide.

Reply:
Sorry for the delay.

The issue with this piece of code is that `GluonCheckSite` is put into the ipk packages as a postinst script; by conditionally changing the definition, you are also changing the package files (which may not be rebuilt when a regular build is run later?).

Optimally, we wouldn't need different logic in `GluonCheckSite` for regular and `push_pkg.sh` builds at all. If it can't be avoided, the condition must be checked at package install time, not build time.

Reply:
Thanks for the feedback.

> Optimally, we wouldn't need different logic in GluonCheckSite for regular and push_pkg.sh builds at all.

I didn't find a way to do this.

> If it can't be avoided, the condition must be checked at package install time, not build time.

I implemented this one.

Reply:
> > Optimally, we wouldn't need different logic in GluonCheckSite for regular and push_pkg.sh builds at all.
> 
> I didn't find a way to do this.

I *think* I have an idea how to achieve this, but let's get https://github.com/freifunk-gluon/gluon/pull/2234 merged first.

Reply:
I have implemented this in #2237.